### DISCUSSION TITLE
Concerns about 2FA OTPs not originating from GitHub

### DISCUSSION THREAD

**Discussion by Praveen-Pandiyan** [community/community#110224] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Bug

### Body

Hello GitHub Community,

I hope this message finds you well. I wanted to bring up a concerning issue that some users, including myself, have been experiencing recently regarding the Two-Factor Authentication (2FA) process on GitHub.

There have been reports of users receiving One-Time Passwords (OTPs) during the GitHub login process that do not appear to be generated by GitHub itself. This has raised concerns about the security and integrity of the 2FA system.

![image](https://github.com/community/community/assets/52589118/88b2527a-ee64-4609-ae7c-d425d70edc32)
![image](https://github.com/community/community/assets/52589118/7cff6613-5735-49a9-93b9-44ab99a625d5)

Initially i ignored these messages, then i tried this OTP after verifying that its not some sort of  phishing attacks. and finally i got logged in to my GitHub account.

If you have experienced similar issues or have insights into the possible causes and solutions, please share your thoughts in this discussion. Additionally, if you work for GitHub or have connections within the GitHub team, your input on the status of investigation and potential resolutions would be highly appreciated.

Thank you for your attention and collaboration.

**Comment by hpsin** [comment, marked as answer]
Thanks for letting us know! 

This is unfortunately your phone carrier reusing or confusing sender identities, and populating the wrong details for the prompt. While it does cause confusion and doesn't look good, it's still a message from GitHub - your carrier is just saying it's from someone else.  We validated that on our side, we are indeed sending the expected text with those OTP codes, and our SMS service (Vonage) indicated that the correct text was passed to Jio (the JD piece in your image). 

The OTP isn't logged to the incorrect company, so no one else is getting this data. We've reached out to our SMS service to see if they can sort something out with Jio though, since it is quite ugly looking.

Phone carriers being susceptible to mistakes and the known security issues with SIM swapping are why we don't recommend SMS and make an effort to promote TOTP instead. We strongly recommend setting up TOTP and a passkey instead, which you can control yourself.

**Comment by Praveen-Pandiyan** [comment]
@hpsin Thanks for the reply,

I wrote to Jio service about this. and closing this discussion. If anyone want to continue feel free to reopen.

**Comment by marcanii** [comment]
responsive #1
